tests/props/enc/custom_props.c

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+
+OE_INFO_SECTION_BEGIN
+volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx = {
+    .header = {.size = sizeof(oe_sgx_enclave_properties_t),
+               .enclave_type = OE_ENCLAVE_TYPE_SGX,
+               .size_settings = {.num_heap_pages = 512,
+                                 .num_stack_pages = 512,
+                                 .num_tcs = 4}},
+    .config = {.product_id = 1111,
+               .security_version = 2222,
+               .attributes = /* These 2 flags cannot be tested as SGX common
```

Comment:
@binxing Why is this disabled (similarly for line 210 in host.c) i.e. what additional errors are you encountering given the changes to sgxload.c and sgxproperties.h already?

Reply:
Copying context from #3430:

> One more problem that can't be addressed by #3526, is that the PROVISION_KEY attribute requires additional authorization by opening /dev/sgx/provision for the current in-tree driver. OE SDK is unaware of the kernel interface so I think it should be captured inside common enclave loader.

The default test configuration that we run in CI/CD still uses out-of-tree drivers, so these tests could still be enabled, no? Broadly speaking, once the OE SDK CI/CD starts to run against the in-tree SGX driver, we would have to configure permissions for all the test apps that used DCAP (since it would require process/user/group permissions) as well, and those changes would be needed for this new test as well, correct?

Is there a tracking item on the Intel side for this? I'm unclear on what change is expected in the enclave common loader for this to work.

Reply:
The error I got was `ENCLAVE_NOT_AUTHORIZED` (`6`). It was due to `-EACCES` returned by the driver. However, I double-checked the common loader source today and figured that it did open `/dev/sgx/provision` whenever `PROVISION_KEY` is requested. What's missing was that my user account didn't belong to `sgx_prv` group. The test case then worked after the group membership was fixed. I have to apologize for the confusion.

I've updated the test case to request `PROVISION_KEY` on the PR branch and now everything is working. Please note that `EINITTOKEN_KEY` won't work (therefore not set/tested in the test case) because the in-tree driver blocks it. But that shouldn't be a practical problem as LE is no longer necessary.

Reply:
@CodeMonkeyLeet, which driver is using in CI/CD?  DCAP driver? If yes and OE has a test enclave requires to access provision key, the CI/CD should update configuration to grant the permission of loading a provision enclave following the DCAP driver [README ](https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/driver/linux#launching-an-enclave-with-provision-bit-set).

Reply:
@lzha101 Thanks for the pointer! OE CI/CD is currently still using the DCAP driver but doesn't use the provisioning key bit directly (it's only used through the PCE/QE as part of attestation). I believe we're still waiting for the availability of the in-tree driver to be available in the ubuntu image before testing the permission grants. It's largely a resource prioritization decision since we're also shifting the technology stack for our CI/CD right now, and we're keeping an eye on the progress of the in-tree driver upstreaming.